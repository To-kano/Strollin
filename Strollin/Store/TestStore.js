import { createStore, combineReducers } from 'redux';
import TestReducer from './Reducers/testReducer';

const rootReducer = combineReducers({
    test: TestReducer
});

export default createStore(rootReducer);
