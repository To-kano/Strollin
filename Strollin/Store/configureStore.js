import { createStore, combineReducers } from 'redux';
import profileReducer from './Reducers/profileReducer';
import galleryReducer from './Reducers/galleryReducer';
import geolocalisationReducer from './Reducers/geolocalisationReducer';
import mapNavigationReducer from './Reducers/mapNavigationReducer';
import chatConversation from './Reducers/conversationReducer';
import socketReducer from './Reducers/socketReducer';
import tendanceReducer from './Reducers/tendanceReducer';
import messageReducer from './Reducers/messageReducer';

const rootReducer = combineReducers({
  profil: profileReducer,
  gallery: galleryReducer,
  position: geolocalisationReducer,
  map: mapNavigationReducer,
  conversation: chatConversation,
  socket: socketReducer,
  tendance: tendanceReducer,
  message: messageReducer
});

export default createStore(rootReducer);
