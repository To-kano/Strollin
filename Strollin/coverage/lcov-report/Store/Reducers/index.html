
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for Store/Reducers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> Store/Reducers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">25.33% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>38/150</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">34.48% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>20/58</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">52.63% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/19</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">25.33% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>38/150</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="conversationReducer.js"><a href="conversationReducer.js.html">conversationReducer.js</a></td>
	<td data-value="20" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="20" class="abs low">4/20</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="20" class="abs low">4/20</td>
	</tr>

<tr>
	<td class="file low" data-value="createConversationReducer.js"><a href="createConversationReducer.js.html">createConversationReducer.js</a></td>
	<td data-value="26.67" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 26%"></div><div class="cover-empty" style="width: 74%"></div></div>
	</td>
	<td data-value="26.67" class="pct low">26.67%</td>
	<td data-value="15" class="abs low">4/15</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="26.67" class="pct low">26.67%</td>
	<td data-value="15" class="abs low">4/15</td>
	</tr>

<tr>
	<td class="file medium" data-value="galleryReducer.js"><a href="galleryReducer.js.html">galleryReducer.js</a></td>
	<td data-value="60" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 60%"></div><div class="cover-empty" style="width: 40%"></div></div>
	</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	</tr>

<tr>
	<td class="file low" data-value="geolocalisationReducer.js"><a href="geolocalisationReducer.js.html">geolocalisationReducer.js</a></td>
	<td data-value="27.27" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 27%"></div><div class="cover-empty" style="width: 73%"></div></div>
	</td>
	<td data-value="27.27" class="pct low">27.27%</td>
	<td data-value="11" class="abs low">3/11</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="27.27" class="pct low">27.27%</td>
	<td data-value="11" class="abs low">3/11</td>
	</tr>

<tr>
	<td class="file low" data-value="mapNavigationReducer.js"><a href="mapNavigationReducer.js.html">mapNavigationReducer.js</a></td>
	<td data-value="9.68" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 9%"></div><div class="cover-empty" style="width: 91%"></div></div>
	</td>
	<td data-value="9.68" class="pct low">9.68%</td>
	<td data-value="31" class="abs low">3/31</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="7" class="abs low">1/7</td>
	<td data-value="9.68" class="pct low">9.68%</td>
	<td data-value="31" class="abs low">3/31</td>
	</tr>

<tr>
	<td class="file medium" data-value="messageReducer.js"><a href="messageReducer.js.html">messageReducer.js</a></td>
	<td data-value="62.5" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 62%"></div><div class="cover-empty" style="width: 38%"></div></div>
	</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	</tr>

<tr>
	<td class="file low" data-value="profileReducer.js"><a href="profileReducer.js.html">profileReducer.js</a></td>
	<td data-value="11.43" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 11%"></div><div class="cover-empty" style="width: 89%"></div></div>
	</td>
	<td data-value="11.43" class="pct low">11.43%</td>
	<td data-value="35" class="abs low">4/35</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="11.43" class="pct low">11.43%</td>
	<td data-value="35" class="abs low">4/35</td>
	</tr>

<tr>
	<td class="file low" data-value="searchReducer.js"><a href="searchReducer.js.html">searchReducer.js</a></td>
	<td data-value="36.36" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 36%"></div><div class="cover-empty" style="width: 64%"></div></div>
	</td>
	<td data-value="36.36" class="pct low">36.36%</td>
	<td data-value="11" class="abs low">4/11</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="36.36" class="pct low">36.36%</td>
	<td data-value="11" class="abs low">4/11</td>
	</tr>

<tr>
	<td class="file medium" data-value="socketReducer.js"><a href="socketReducer.js.html">socketReducer.js</a></td>
	<td data-value="66.67" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 66%"></div><div class="cover-empty" style="width: 34%"></div></div>
	</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	</tr>

<tr>
	<td class="file medium" data-value="tendanceReducer.js"><a href="tendanceReducer.js.html">tendanceReducer.js</a></td>
	<td data-value="50" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 50%"></div><div class="cover-empty" style="width: 50%"></div></div>
	</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Jan 04 2021 15:58:26 GMT+0100 (Central European Standard Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    