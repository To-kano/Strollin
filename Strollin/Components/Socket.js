import React, { useEffect, useState, useContext} from 'react';
import { connect } from 'react-redux';
import { StyleSheet, TextInput, View } from 'react-native';
import socketIOClient from 'socket.io-client';
import { createContext } from "react";
import Store from '../Store/configureStore';
import profileReducer from '../Store/Reducers/profileReducer';
import { IP_SERVER, PORT_SERVER } from '../env/Environement';

const SocketContext = createContext();
const ENDPOINT = `http://${IP_SERVER}:${PORT_SERVER}`;//3000 pour Tony
import AsyncStorage from '@react-native-community/async-storage';



export async function getProfilCache(dispatch) {
  try {
    let jsonValue = await AsyncStorage.getItem('cache_profile');

    if (jsonValue) {
      jsonValue = JSON.parse(jsonValue);

      const action = { type: 'SET_USER', value: jsonValue };
      dispatch(action);
    }
  } catch (e) {
  //console.log('echec store profile ', e);
  }
};

function Socket({children, profil, dispatch}) {
  const [socket, setSocket] = useState(null);

  const store = Store.getState();

  if (socket == null) {
    setSocket(socketIOClient(ENDPOINT));
  }

  if (!profil.access_token) {
    getProfilCache(dispatch);
    //console.log('nav props: ', profil);
  }

  useEffect(() => {
    if (socket != null) {
      socket.on('receiveMessage', (data) => {
    //console.log("received msg" + JSON.stringify(data));

      const action = { type: 'ADD_MESSAGE', value: data };
      Store.dispatch(action);

      const action2 = {
        type: 'ADD_MESSAGE_ID',
        value: { id: data.conversation, message_id : data.id}
      };
      Store.dispatch(action2);

      });
      socket.on('identification', (data) => {
        //console.log("identification = ", data);
      });
      socket.on('newConversation', (data) => {
        const store = Store.getState();

        const action = { type: 'ADD_CONVERSATION', value: data };
        Store.dispatch(action);
        const action2 = { type: 'RESET_PARTICIPANT_OF_CONVERSATION'};
        Store.dispatch(action2);
        const action3 = { type: 'ADD_TO_SEARCH_CONV', value: data };
        Store.dispatch(action3);
      });
    }
  }, [socket]);

  useEffect(() => {
    if (profil.access_token) {
      //console.log("access_token = ", profil.access_token);
      socket.emit('login', { access_token: profil.access_token });
      socket.emit('sendMessage', "hey Pierre!");
    }

  }, [profil.access_token])

  const sendMessage = (message) => {
    //console.log('sending message ', message);


    socket.emit('sendMessage', { access_token: store.profil.access_token,
      conversation: store.conversation.currentConversation, type: "message", message: message});
  };

  const sendImage = (image) => {
  //console.log('sending image', image);

    socket.emit('sendMessage', { access_token: store.profil.access_token,
      conversation: store.conversation.currentConversation, type: "image", message: image});
  };

  const sendCourse = (courseId) => {
  //console.log('sending course', courseId);

    socket.emit('sendMessage', { access_token: store.profil.access_token,
      conversation: store.conversation.currentConversation, type: "course", message: courseId});
  };

  const createConversation = (participantsID) => {
  //console.log('creating conversation', participantsID);
    let convName = store.profil.pseudo;

    for (let i in participantsID) {
      let tmp = ", " + store.profil.friends_pseudo_list[participantsID[i]];
      convName += tmp;
    }
    var object = { access_token: store.profil.access_token, participants: participantsID, name: convName }
    socket.emit('createConversation', object);
  };

  return (
    <SocketContext.Provider value={{ socket, sendMessage, createConversation, sendImage, sendCourse }}>
      {children}
    </SocketContext.Provider>
  );
}

export const contextSocket = () => useContext(SocketContext);

const mapStateToProps = (state) => state;
export default connect(mapStateToProps)(Socket);
