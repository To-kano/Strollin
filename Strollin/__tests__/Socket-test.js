import React from 'react';
import {shallow} from 'enzyme';
import Socket, { getProfilCache } from '../Components/Socket';

import Store from '../Store/configureStore';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';


const profilTest = {
    access_token : "vjwhvjhkvjhwkvhjgdkw"
}

const dispatchTest = (test) => jest.fn();

describe('Socket', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(
            <Provider store={Store}>
                <Socket profil={profilTest} dispatch={dispatchTest}/>
            </Provider>)
            expect(component).toMatchSnapshot()
        });

        test('test render socket', () => {
            renderer.create(
                <Provider store={Store}>
                    <Socket profil={profilTest} dispatch={dispatchTest}/>
                </Provider>)
          })

          it('test getProfileCace',async () => {

                await getProfilCache(dispatchTest);
          })
    });
});