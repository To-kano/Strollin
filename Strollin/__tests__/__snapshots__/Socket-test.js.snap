// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Socket Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
